addons/point_of_sale/models/pos_order.py

```diff
@@ -514,6 +514,7 @@ def _reconcile_payments(self):
                 # Whereas freight returns won't be
                 # "c'est la vie..."
                 aml.reconcile()
+                aml.mapped('payment_id').write({'partner_type': 'customer'})
```

Comment:
not the right place to do so... partner_type should be given at the payment creation.